src/webgl-preview.ts

```diff
@@ -460,13 +543,21 @@ export class WebGLPreview {
     this.scene.add(geometryBox);
   }
 
+  // reset parser & processing state
   clear(): void {
+    this.resetState();
+    this.parser = new Parser(this.minLayerThreshold);
+  }
+
+  // reset processing state
+  resetState(): void {
     this.startLayer = 1;
     this.endLayer = Infinity;
     this.singleLayerMode = false;
-    this.parser = new Parser(this.minLayerThreshold);
     this.beyondFirstMove = false;
-    this.state = { x: 0, y: 0, z: 0, r: 0, e: 0, i: 0, j: 0, t: 0 };
+    this.state = State.initial;
+    this.prevState = undefined;
+    this.prevLayerIndex = undefined;
```

Comment:
Same question about keeping this private. State is an internal concept that should not be messed with.